src/api/l_event.c

```diff
@@ -65,6 +65,23 @@ void luax_checkvariant(lua_State* L, int index, Variant* variant) {
       break;
     }
 
+    case LUA_TTABLE:
+      if (index < 0) {
+        index = lua_gettop(L) + index + 1;
+      }
+      size_t length = luax_len(L, index);
+      variant->type = TYPE_TABLE;
+      variant->value.table.length = length;
+
+      Variant* list = lovrMalloc(length * sizeof(Variant));
```

Comment:
This should be freed in `lovrVariantDestroy` (and also recurse and destroy the children variants).